### PR TITLE
fix: data api loading states and repetitive calls

### PR DESCRIPTION
**Pull request by cartcrom**
Used new Suspense wrappers to replace default Suspense spinner, consolidated getting price change delta into toptoken call instead of individual row calls

### CODE REVIEW

File: src/components/Tokens/TokenDetails/TokenDetail.tsx [hunk 1]
```diff
@@ -293,128 +213,72 @@ export default function LoadedTokenDetail({ address }: { address: string }) {
...
+        <StatsSection>
+          <StatPair>
+            <Stat>
+              Market cap
+              <StatPrice>
+                {tokenDetailData.marketCap?.value ? formatDollarAmount(tokenDetailData.marketCap?.value) : '-'}
+              </StatPrice>
+            </Stat>
+            <Stat>
+              24H volume
```
Comment (cmcewen) [inline src/components/Tokens/TokenDetails/TokenDetail.tsx:246]: does this need to be translated? or are we skipping that?

File: src/components/Tokens/TokenTable/TokenRow.tsx [hunk 2]
```diff
@@ -457,15 +456,17 @@ export default function LoadedRow({
...
-  const pricePoints: PricePoint[] = useTokenPriceQuery(tokenAddress, timePeriod, 'ETHEREUM').filter(
-    (p): p is PricePoint => Boolean(p && p.value)
-  )
-  const hasData = pricePoints.length !== 0
-
-  /* TODO: Implement API calls & cache to use here */
-  const startingPrice = hasData ? pricePoints[0] : DATA_EMPTY
-  const endingPrice = hasData ? pricePoints[pricePoints.length - 1] : DATA_EMPTY
-  const [delta, arrow] = getDelta(startingPrice.value, endingPrice.value)
+  // const pricePoints: PricePoint[] = useTokenPriceQuery(tokenAddress, timePeriod, 'ETHEREUM').filter(
```
Comment (cmcewen) [inline src/components/Tokens/TokenTable/TokenRow.tsx:459]: is this just to make your todo easier to pick up? or can it be deleted

File: src/pages/App.tsx [hunk 3]
```diff
@@ -160,8 +161,22 @@ export default function App() {
...
+                        element={
+                          <Suspense fallback={<LoadingTokens />}>
+                            <Tokens />
+                          </Suspense>
+                        }
+                      />
+                      <Route
+                        path="/tokens/:tokenAddress"
+                        element={
+                          <Suspense fallback={<LoadingTokenDetails />}>
```
Comment (cmcewen) [inline src/pages/App.tsx:175]: coooool

File: src/pages/TokenDetails/index.tsx [hunk 4]
```diff
@@ -83,15 +81,11 @@ export default function TokenDetails() {
...
-    // TODO: handle no address / invalid address cases
-    tokenDetail = 'invalid token'
-  } else if (loading) {
-    tokenDetail = <LoadingTokenDetail />
-  } else {
-    tokenDetail = <TokenDetail address={tokenAddress} />
-  }
+  useEffect(() => {
+    return function cleanup() {
+      console.log('Unmounted higher')
```
Comment (cmcewen) [inline src/pages/TokenDetails/index.tsx:86]: looks like you don't need this?

**Comment by cmcewen** [comment]
also note you have a file conflict to resolve

File: src/components/Tokens/TokenDetails/TokenDetail.tsx (hunk 1 above)
Comment (cartcrom) [reply src/components/Tokens/TokenDetails/TokenDetail.tsx:246]: How should something like '24H' be translated? or should I just translate volume?

Comment (cartcrom) [reply src/components/Tokens/TokenDetails/TokenDetail.tsx:246]: I'm thinkin maybe a sep task to go into tokens/ token details and figure out what/should be <Trans>'d

**Review by cmcewen — APPROVED** [review]
could you make a follow up task to check on i18n?

**Pull request merged**